balancer_wrapper.go

```diff
@@ -277,10 +277,17 @@ type healthData struct {
 	// to the LB policy. This is stored to avoid sending updates when the
 	// SubConn has already exited connectivity state READY.
 	connectivityState connectivity.State
+	// closeHealthProducer stores function to close the ref counted health
+	// producer. The health producer is automatically closed when the SubConn
+	// state changes.
```

Comment:
We are currently only calling this when `RegisterHealthListener` and not when the subconn's connectivity state changes. Am I missing something?

Reply:
The producers are closed automatically on every subchnnel state change here: https://github.com/grpc/grpc-go/blob/b1f70ce0555fffe2a384969eacf8beaa73454695/balancer_wrapper.go#L288-L294

Closing them explicitly isn't required.

Reply:
Ack, thanks.